src/bin/tester.rs

```diff
@@ -163,92 +247,85 @@ fn main() {
-
-            let mut stdout = io::stdout();
+                .filter(|entry| entry.path().extension().map_or(false, |ext| ext == "rs"))
+                .filter(|entry| !BLACKLIST.iter().any(|&b| entry.path().ends_with(b)));
 
             // Go through all the files and try to parse each of them.
-            for file in files {
-                let path = file.into_path();
-
-                total_count += 1;
-                if !verbose {
-                    // Limit the compact output to 80 columns wide.
-                    if total_count % 80 == 0 {
-                        println!("");
-                    }
-                }
-
-                // HACK(eddyb) avoid parsing some files that hit
-                // `lykenware/gll` worst-cases (many GBs of RAM usage)
-                // FIXME(eddyb) fix the problems (e.g. implement GC).
-                const BLACKLIST: &[&str] = &[
-                    "libcore/unicode/tables.rs",
-                    "issues/issue-29466.rs",
-                    "issues/issue-29227.rs",
-                ];
-                if BLACKLIST.iter().any(|&b| path.ends_with(b)) {
-                    if verbose {
-                        eprintln!("{}: SKIP (blacklisted)...", path.display());
-                    } else {
-                        print!("S");
-                        stdout.flush().unwrap();
-                    }
-                    continue;
-                }
 
-                // Indicate the current file being parsed in verbose mode.
-                // This can be used to find files to blacklist (see above).
-                if verbose {
-                    eprint!("{}...\r", path.display());
-                }
-
-                parse_file_with(&path, |result| {
-                    // Increment counters and figure out the character to print.
-                    let mut ambiguity_result = Ok(());
-                    let (status, count) = match result {
-                        Ok(handle) => {
-                            ambiguity_result = ambiguity_check(handle);
-                            if ambiguity_result.is_ok() {
-                                ('.', &mut unambiguous_count)
-                            } else {
-                                ('-', &mut ambiguous_count)
+            let mut counters: Counters = files
+                .par_bridge()
+                .map(|f| process(f, verbose))
+                .fold(
+                    || Counters {
+                        total_count: 0,
+                        ambiguous_count: 0,
+                        unambiguous_count: 0,
+                        too_short_count: 0,
+                        no_parse_count: 0,
+                    },
+                    |mut acc, x| {
+                        acc.total_count += 1;
+                        match x {
+                            ParseResult::Ambiguous => {
+                                acc.ambiguous_count += 1;
```

Comment:
Not sure if better, but you could replace this with `=> &mut acc.ambiguous_count,` (for each arm + field) and then increment the returned field.

Reply:
the original code was doing this and it took me a while to figure out what it was doing, hence I've opted for the more explicit version